src/components/tooltip/tooltip.js

```diff
@@ -10,6 +35,159 @@ import {h, Component} from 'preact';
+   * @param {*} props props
+   * @memberof Tooltip
+   */
+  constructor(props: Object) {
+    super(props);
+    this.setState({showTooltip: false, marginTop: 0, marginLeft: 0});
+  }
+
+  /**
+   * gets the tooltip position based on component prop
+   *
+   * @returns {void}
+   * @memberof Tooltip
+   */
+  setMargins() {
+    const tooltipBoundingRect = this.textElement ? this.textElement.getBoundingClientRect() : null;
+    switch (this.state.type) {
+      case ToolTipType.Top:
+        this.setState({marginLeft: tooltipBoundingRect ? -tooltipBoundingRect.width / 2 : 0, marginTop: 0});
+        break;
+      case ToolTipType.TopLeft:
+        this.setState({marginLeft: tooltipBoundingRect ? -tooltipBoundingRect.width + 16 : 0, marginTop: 0});
+        break;
+      case ToolTipType.TopRight:
+        this.setState({marginLeft: -16, marginTop: 0});
+        break;
+      case ToolTipType.Left:
+        this.setState({marginTop: tooltipBoundingRect ? -tooltipBoundingRect.height / 2 : 0, marginLeft: 0});
+        break;
+      case ToolTipType.Right:
+        this.setState({marginTop: tooltipBoundingRect ? -tooltipBoundingRect.height / 2 : 0, marginLeft: 0});
+        break;
+      case ToolTipType.Bottom:
+        this.setState({marginLeft: tooltipBoundingRect ? -tooltipBoundingRect.width / 2 : 0, marginTop: 0});
+        break;
+      case ToolTipType.BottomRight:
+        this.setState({marginLeft: -16, marginTop: 0});
+        break;
+      case ToolTipType.BottomLeft:
+        this.setState({marginLeft: tooltipBoundingRect ? -tooltipBoundingRect.width + 16 : 0, marginTop: 0});
+        break;
+    }
+  }
+
+  /**
+   * on mouse over handler.
+   * @memberof Tooltip
+   * @returns {void}
+   */
+  onMouseOver(): void {
+    if (this._hoverTimeout) {
+      clearTimeout(this._hoverTimeout);
+      this._hoverTimeout = null;
+    }
+    this._hoverTimeout = setTimeout(() => {
+      this.setState({showTooltip: true});
+    }, TOOLTIP_SHOW_TIMEOUT);
+  }
+
+  /**
+   * on mouse leave handler.
+   * @memberof Tooltip
+   * @returns {void}
+   */
+  onMouseLeave(): void {
+    this.setState({showTooltip: false});
+    clearTimeout(this._hoverTimeout);
+    this._hoverTimeout = null;
+  }
+
+  /**
+   * brings another tooltip type which hasn't been marked as invalid
+   * @memberof Tooltip
+   * @returns {string} tooltip type
+   */
+  getAlternateType(): string {
+    for (const key in ToolTipType) {
+      const toolTipType = ToolTipType[key];
+      if (this.nonValidTypes.indexOf(toolTipType) === -1) {
+        return toolTipType;
+      }
+    }
+    return this.state.type;
+  }
+
+  /**
+   * checks if the current tooltip type is within the player boundaries
+   * @memberof Tooltip
+   * @returns {string} is in boundaries
+   */
+  isToolTipInBoundaries(): boolean {
+    const tooltipBoundingRect = this.textElement.getBoundingClientRect();
+    const playerContainerRect = this.props.playerClientRect;
+    const MARGIN = 10;
```

Comment:
You have here constant like margin, that you are using it on css as well.
Could it be in css?

Reply:
magic numbers should be in the top of the file as CAPITAL snake case

Reply:
This is not the same margin as in the css - this represents the player margins (So the tooltip will reach until the last pixels). I renamed it to PLAYER_MARGIN and moved to the top like Oren mentioned.